README.md: Rewrite of branch and tag policies

frznlogic opened the pull request:
The release policy of the GENIVI Baseline will be changed and we want to
continue doing bi-annual releases in meta-ivi. These changes should
reflect that and further explain how our releases and branches relates
to the GENIVI Baseline.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

Commit:
README.md: Rewrite of branch and tag policies

The release policy of the GENIVI Baseline will be changed and we want to
continue doing bi-annual releases in meta-ivi. These changes should
reflect that and further explain how our releases and branches relates
to the GENIVI Baseline.

Signed-off-by: Oscar Andreasson <oscar.andreasson@pelagicore.com>

gunnarx commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
+  numbers in sync, we will release 14.0.0 together with the Compliance
+  Specification and 14.50.0 half way to 15.0.0. 
+  - Note: Updates to the 14.0.0 release would then become 14.0.1 and/or 14.1.0, which
+    can be released __after__ 14.50.0.
```
I'd change to "(Note that such patch/minor updates could be released after 14.50.0 in certain cases)" or maybe just drop it?  It's more of special case I think.

frznlogic replied:
Fixed I think.

gunnarx commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
+  Specification and 14.50.0 half way to 15.0.0. 
+  - Note: Updates to the 14.0.0 release would then become 14.0.1 and/or 14.1.0, which
+    can be released __after__ 14.50.0.
+- Somewhere near a new major release of the GENIVI Compliance specification, a
```
Update to what we decided (14.0.x-rocko).
I think the repeated sections "Somewhere..." and "In the middle..." are perhaps a bit redundant/repetitive.

This whole section can be simplified a bit, just by saying
"Maintenance branches for each release will be named ...."

frznlogic replied:
Fixed.

gunnarx commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
 
+Tag Policy
+----------
+- GENIVI releases use release names called P-0.1, P-0.2, P-1.0 indicating
```
All these details might not be necessary here.  It's details about the process between the compliance team responsibles and the baseline...  I think it might enough to explain that P-something are "internal" tags corresponding to the "P" release.  Maybe just instruct "most users" to rely on the semver tags?

frznlogic replied:
Mostly removed (I keep a short bullet that we have internal releases)

gunnarx commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
+  - P-1.0 is equivalent to 14.0.0 and should be the final release. 
+- Meta-ivi creates tags with the GENIVI release naming scheme upon releasing
+  new versions.
+- Meta-ivi creates tags based on the version number in accordance with this
```
Agree with these two last points, good to keep them.

frznlogic replied:
Fixed.

gunnarx commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
+  new versions.
+- Meta-ivi creates tags based on the version number in accordance with this
+  policy.
+- Meta-ivi creates a Github release based on the version number in accordance
```
Also this, good to keep.  The text is a bit repetitive.  I think you can just combine it to be a bit less clunky.

Proposal:  -- "creates a tag and a corresponding GitHub release..." 
"named according to the version policy"

frznlogic replied:
Fixed.

slawr commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
 
+Tag Policy
+----------
+- GENIVI releases use release names called P-0.1, P-0.2, P-1.0 indicating
```
I suggest reversing the tag policy section and leading with a description of the 'external' semver release names, i.e. the stable releases consumed by the general user.

It then can go onto describe the relationship between the internal A-x.x early dev releases and the external semver names. Doing that under some kind of sub-heading about early releases may help mitigate possible confusion about dual naming.

This all assumes the conclusions of the recent discussion about A-x.x naming still stand.

frznlogic replied:
Removed, just kept a short note about genivi internal tags.

slawr commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
+  it's a pre-release and is work in progress, 0.2 signifies that the Compliance
+  Specification should be at a point where the fundamentals are in place and
+  1.0 is a refinement on 0.2 where the specifications are finalized. 
+  - P-0.1 is equivalent to 14.0.0-rc.1
```
I find '-rc' a little troublesome for the reasons brought up in the A-x.x naming discussion that the early releases might not fit the commonly known definition of a 'release candidate'. They can be very early rough snapshots along the way releases. I can't recall what the conclusion was there. I would need to go back through the BIT minutes.

frznlogic replied:
Gone.

slawr commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
+  - Note: Updates to the 14.0.0 release would then become 14.0.1 and/or 14.1.0, which
+    can be released __after__ 14.50.0.
+- Somewhere near a new major release of the GENIVI Compliance specification, a
+  numbered release branch (e.g. 14.0) is created from master, and the new
```
Minor point for line 24: I thought we favoured 14.x rather than 14.0.

frznlogic replied:
We did. I hope I got all of them correct now.

slawr commented on README.md:
```diff
@@ -9,24 +9,48 @@ Branch Policy
+  applicable to a particular release only.
+- Major numbers are stepped up every new release of the GENIVI Compliance
+  Specification.  The release schedule is driven by a time plan. Use [semantic
+  versioning](http://semver.org/) for minor and patch numbers.
```
Minor point: After the full lines of the previous sentences the shortened English of line 16 is jarring.

slawr reviewed (commented):
Some minor points along with a larger one on the tag section. I didn't add any more as an update is expected based on the conversation in last weeks BIT call.

Commit:
README.md: Final tweaks on branch and tag policies

Reorder the branch policy statements in (I think) a more logical order
- first principles (cherry-pick), then versioning, etc.  Minor word
changes to make the text shorter and clearer.  Add note about Yocto/Poky
upgrades, as introduction to the next point which talks about
-rocko/-sumo branch names.  Remove vexpressa9 from list of tested
platforms.  Add a note on how other hardware is supported.

In other sections, minor word changes and whitespace.

Signed-off-by: Gunnar Andersson <gandersson@genivi.org>

gunnarx commented:
Squashed your two commits (the last one was a minor spelling fix).
I then added my own and kept it separate since you might want to see what I changed.

gunnarx commented:
Merged
